beta.3 (#1635) breaks custom binaries (on Alpine)

## Bug description

Before beta3, custom binaries could be built for Alpine and use with ENV variables. A recent change with #1635 breaks this functionality by bombing out of the npm process if the environment is Alpine, regardless of if custom binaries exist, or if Alpine was patched to support glibc (as raised in #702).

## How to reproduce

Build binaries for musl (Alpine), set the ENV variables for custom binaries, and attempt an `npm install`.
See https://gitlab.com/recipe-cookbook/backend/-/jobs/521166425 for repro build failure

## Expected behavior

Process to complete normally. I would expect that if the ENV binary fields are used (and point to real files), the short-circuit exit on Alpine would not happen. It's not important (to me), but for others in #702, if a valid glibc environment exists, then the platform is valid and can run the stock binaries and should also not exit out.

## Prisma information

Prisma 2.0.0-beta.3
Schema: https://gitlab.com/recipe-cookbook/backend/-/blob/prisma/beta3/prisma/schema.prisma

## Environment & setup

- OS: Alpine Linux
- Database: PostgreSQL
- Prisma version: 2.0.0-beta.3
- Node.js version: v13.12.0

@Jolg42 - This is not yet resolved, build pipeline still throws an error on alpha 1129: https://gitlab.com/recipe-cookbook/backend/-/jobs/526450525 - specifically:

```
 #11 38.88 > @prisma/client@2.0.0-alpha.1129 postinstall /app/node_modules/@prisma/client
#11 38.88 > node scripts/postinstall.js
#11 38.88 
#11 40.43 Warning Precompiled binaries are not available for Alpine.
#11 40.54 Warning Precompiled binaries are not available for Alpine.
#11 40.54 Error: Unknown binaryTargets musl-openssl-1.1.x
#11 41.00 
#11 41.00 > cookbook-backend@undefined postinstall /app
#11 41.00 > npm -s run generate
#11 41.00 
#11 43.75 Warning Precompiled binaries are not available for Alpine.
#11 43.85 Warning Precompiled binaries are not available for Alpine.
#11 43.85 Error: Unknown binaryTargets musl-openssl-1.1.x
#11 43.92 npm ERR! code ELIFECYCLE
#11 43.93 npm ERR! errno 1
#11 43.94 npm ERR! cookbook-backend@undefined postinstall: `npm -s run generate`
#11 43.94 npm ERR! Exit status 1
#11 43.94 npm ERR! 
#11 43.95 npm ERR! Failed at the cookbook-backend@undefined postinstall script.
#11 43.95 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
#11 44.04 
#11 44.05 npm ERR! A complete log of this run can be found in:
#11 44.05 npm ERR!     /home/node/.npm/_logs/2020-04-25T02_20_00_431Z-debug.log
```

The error occurs during prisma generation:

```
> npm -s run clean && npm -s run generate && tsc

2020-04-28T23:21:07.519Z prisma {
  argv: [
    '/usr/local/bin/node',
    '/app/node_modules/.bin/prisma',
    'generate'
  ]
}
2020-04-28T23:21:07.523Z prisma Environment variables not loaded
2020-04-28T23:21:07.567Z getos { version: '' }
2020-04-28T23:21:07.568Z getos { ls: '' }
2020-04-28T23:21:07.570Z getos { platform: 'linux', libssl: undefined }
2020-04-28T23:21:07.585Z getos { version: '' }
2020-04-28T23:21:07.586Z getos { ls: '' }
Warning Precompiled binaries are not available for Alpine.
2020-04-28T23:21:07.593Z download Using env var PRISMA_QUERY_ENGINE_BINARY for binary query-engine, which points to /app/alpine-x86_64/query-engine
2020-04-28T23:21:07.593Z download Using env var PRISMA_MIGRATION_ENGINE_BINARY for binary migration-engine, which points to /app/alpine-x86_64/migration-engine
2020-04-28T23:21:07.593Z download Using env var PRISMA_INTROSPECTION_ENGINE_BINARY for binary introspection-engine, which points to /app/alpine-x86_64/introspection-engine
2020-04-28T23:21:07.598Z download { fileExists: false, cachedFile: null }
2020-04-28T23:21:07.598Z download { needsToBeDownloaded: true }
2020-04-28T23:21:07.599Z download { fileExists: false, cachedFile: null }
2020-04-28T23:21:07.599Z download { needsToBeDownloaded: true }
2020-04-28T23:21:07.600Z download { fileExists: false, cachedFile: null }
2020-04-28T23:21:07.600Z download { needsToBeDownloaded: true }
2020-04-28T23:21:07.620Z getos { version: '' }
2020-04-28T23:21:07.620Z getos { ls: '' }
2020-04-28T23:21:07.629Z getos { platform: 'linux', libssl: undefined }
2020-04-28T23:21:07.649Z getos { version: '' }
2020-04-28T23:21:07.649Z getos { ls: '' }
2020-04-28T23:21:07.651Z getos { platform: 'linux', libssl: undefined }
2020-04-28T23:21:07.667Z getos { version: '' }
2020-04-28T23:21:07.667Z getos { ls: '' }
Warning Precompiled binaries are not available for Alpine.
Error: Error: Unknown binaryTargets musl-openssl-1.1.x
    at Object.download (/app/node_modules/@prisma/cli/build/index.js:2:2331638)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! cookbook-backend@ build: `npm -s run clean && npm -s run generate && tsc`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the cookbook-backend@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-04-28T23_21_07_797Z-debug.log
The command '/bin/sh -c ./setup-env.sh npm run build' returned a non-zero code: 1
```

I've made a repro repo here: https://github.com/radicand/prisma-issue-2266

--- a/src/packages/engine-core/src/NodeEngine.ts
+++ b/src/packages/engine-core/src/NodeEngine.ts
@@ -57,6 +57,8 @@ const knownPlatforms: Platform[] = [
   'debian-openssl-1.1.x',
   'rhel-openssl-1.0.x',
   'rhel-openssl-1.1.x',
+  'musl-openssl-1.0.x',
+  'musl-openssl-1.1.x',
   'windows',
 ]
 

--- a/src/packages/fetch-engine/scripts/test.ts
+++ b/src/packages/fetch-engine/scripts/test.ts
@@ -13,6 +13,8 @@ download({
     'debian-openssl-1.1.x',
     'rhel-openssl-1.0.x',
     'rhel-openssl-1.1.x',
+    'musl-openssl-1.0.x',
+    'musl-openssl-1.1.x',
   ],
   showProgress: true,
 })

--- a/src/packages/get-platform/src/platforms.ts
+++ b/src/packages/get-platform/src/platforms.ts
@@ -5,6 +5,8 @@ export type Platform =
   | 'debian-openssl-1.1.x'
   | 'rhel-openssl-1.0.x'
   | 'rhel-openssl-1.1.x'
+  | 'musl-openssl-1.0.x'
+  | 'musl-openssl-1.1.x'
   | 'windows'
   | 'freebsd'
   | 'openbsd'
@@ -16,6 +18,8 @@ export const platforms = [
   'debian-openssl-1.1.x',
   'rhel-openssl-1.0.x',
   'rhel-openssl-1.1.x',
+  'musl-openssl-1.0.x',
+  'musl-openssl-1.1.x',
   'windows',
 ]
 

--- a/src/packages/sdk/src/getGenerators.ts
+++ b/src/packages/sdk/src/getGenerators.ts
@@ -286,6 +286,8 @@ export const knownBinaryTargets: Platform[] = [
   'debian-openssl-1.1.x',
   'rhel-openssl-1.0.x',
   'rhel-openssl-1.1.x',
+  'musl-openssl-1.0.x',
+  'musl-openssl-1.1.x',
   'windows',
 ]
 

